src/main/java/gregtech/common/mui/widget/GTFluidSlot.java

```diff
@@ -79,31 +78,54 @@ public GTFluidSlot syncHandler(GTFluidSyncHandler syncHandler) {
         return this;
     }
 
+    public GTFluidSlot showAmount(boolean showAmount) {
+        this.showAmount = showAmount;
+        return this;
+    }
+
+    public GTFluidSlot disableBackground() {
+        this.disableBackground = true;
+        return this;
+    }
+
     @Override
-    public void draw(GuiContext context, WidgetTheme widgetTheme) {
+    public boolean isValidSyncHandler(SyncHandler syncHandler) {
+        return syncHandler instanceof GTFluidSyncHandler;
+    }
+
+    @Override
+    public void drawBackground(ModularGuiContext context, WidgetTheme widgetTheme) {
+        if (disableBackground) return;
+        super.drawBackground(context, widgetTheme);
+    }
+
+    @Override
+    public void draw(ModularGuiContext context, WidgetTheme widgetTheme) {
         FluidStack content = this.syncHandler.getFluid();
-        if (content != null) {
-            GuiDraw.drawFluidTexture(content, 1, 1, getArea().w() - 2, getArea().h() - 2, 0);
 
+        GuiDraw.drawFluidTexture(content, 1, 1, getArea().w() - 2, getArea().h() - 2, 0);
```

Comment:
Why did we move this draw above the check for if the content was null?

Reply:
the method already has a null check within it